internal/provider/schema_gitlab_project_milestone.go

```diff
@@ -0,0 +1,127 @@
+			Type:             schema.TypeString,
+			Optional:         true,
+			ValidateDiagFunc: isISO6801Date,
+		},
+		// NOTE: not part of `CREATE`, but part of `UPDATE` with the `state_event` field.
+		"state": {
+			Description:      fmt.Sprintf("The state of the milestone. Valid values are: %s.", renderValueListForDocs(validMilestoneStates)),
+			Type:             schema.TypeString,
+			Optional:         true,
+			Default:          "active",
+			ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice(validMilestoneStates, false)),
+		},
+		"created_at": {
+			Description: "The time of creation of the milestone. Date time string, ISO 8601 formatted, for example 2016-03-11T03:45:40Z.",
+			Type:        schema.TypeString,
+			Optional:    true,
+			Computed:    true,
+			// NOTE: since RFC3339 is pretty much a subset of ISO8601 and actually expected by GitLab,
+			//       we use it here to avoid having to parse the string ourselves.
+			ValidateDiagFunc: validation.ToDiagFunc(validation.IsRFC3339Time),
+		},
+		"updated_at": {
+			Description: "The last update time of the milestone. Date time string, ISO 8601 formatted, for example 2016-03-11T03:45:40Z.",
+			Type:        schema.TypeString,
+			Optional:    true,
+			Computed:    true,
+			// NOTE: since RFC3339 is pretty much a subset of ISO8601 and actually expected by GitLab,
+			//       we use it here to avoid having to parse the string ourselves.
+			ValidateDiagFunc: validation.ToDiagFunc(validation.IsRFC3339Time),
+		},
+		"expired": {
+			Description: "Bool, true if milestore expired.",
+			Type:        schema.TypeBool,
+			Computed:    true,
+		},
+		"iid": {
+			Description: "The ID of the project's milestone.",
+			Type:        schema.TypeInt,
+			Computed:    true,
+		},
+		"milestone_id": {
+			Description: "The instance-wide ID of the project’s milestone.",
+			Type:        schema.TypeInt,
+			Computed:    true,
+		},
+		"project_id": {
+			Description: "The project ID of milestone.",
+			Type:        schema.TypeInt,
+			Computed:    true,
+		},
+		"web_url": {
+			Description: "The web URL of the milestone.",
+			Type:        schema.TypeString,
+			Computed:    true,
+		},
+	}
+}
+
+func gitlabProjectMilestoneToStateMap(project string, milestone *gitlab.Milestone) map[string]interface{} {
+	stateMap := make(map[string]interface{})
+	stateMap["iid"] = milestone.IID
+	stateMap["milestone_id"] = milestone.ID
+	stateMap["project"] = project
+	stateMap["project_id"] = milestone.ProjectID
+	stateMap["title"] = milestone.Title
+	stateMap["description"] = milestone.Description
+	if milestone.DueDate != nil {
+		stateMap["due_date"] = milestone.DueDate.String()
+	} else {
+		stateMap["due_date"] = nil
+	}
+	if milestone.StartDate != nil {
+		stateMap["start_date"] = milestone.StartDate.String()
+	} else {
+		stateMap["start_date"] = nil
+	}
+	stateMap["updated_at"] = milestone.UpdatedAt.Format(time.RFC3339)
```

Comment:
I think that both the `milestones.UpdatedAt` and `milestone.CreatedAt` fields can technically be `nil` (because of their `*time.Time` type. 

However, probably only `milestone.UpdatedAt` ever will be, right?

Reply:
Mmm, I doubt. But yes, as "backup strategy" I can add if condition.

Reply:
Yes, you are right. Strangely the API returns the same `updated_at` as the `created_at` right after creation 🤷 

So, either way is fine for me ;) 

Sorry for the confusion ✌️